spec/lib/workflow_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe Workflow do
+  let(:adapter)       { stub(:password => 'password', :username => 'username') }
+  let(:client)        { stub(:adapter => adapter) }
+  let(:workflow_name) { 'enrollment' }
+  let(:data)          { { customer_id: SecureRandom.uuid } }
+  let(:workflow)      { described_class.new(client:  client,
+                                            name:    workflow_name,
+                                            data:    data ) }
+
+  let(:jwt) { JWT.encode(
+                {
+                  workflow:       workflow_name,
+                  application_id: client.adapter.username,
+                  iat:            Time.now.to_i,
+                  data:           data
+                  },
+                  client.adapter.password
+              )
+            }
+
+  describe "#token" do
+    it 'encodes a JWT' do
+      workflow.token.should eql(jwt)
```

Comment:
it's probably better to decode the token provided by your production code to test it in the test, otherwise you're just duplicating the implementation in the test but not proving that the token decodes correctly. I think a test like this would be better:

```
decoded_token = JWT.decode...
decoded_token["workflow"].should == ...
decoded_token["application_id"].should == ...
...
```